force-app/main/default/classes/DuplicateController.cls

```diff
@@ -0,0 +1,72 @@
+public with sharing class DuplicateController {
+    @AuraEnabled(Cacheable = true)
+    public static List<SObject> findDuplicates(String idFromPage) {
+        Id recordId = Id.valueOf(idFromPage);
+        String objName = recordId.getSObjectType().getDescribe().getName();
+        List<DuplicateRecordSet> duplicateRecordSet = getRecordSetList(recordId);
+
+        List<DuplicateRecordItem> itemsList = [SELECT recordId FROM DuplicateRecordItem WHERE DuplicateRecordSetId =: duplicateRecordSet];
+
+        List<String> recordIds = new List<String>();
+        for(DuplicateRecordItem item : itemsList){
+            recordIds.add(item.recordId);
+        }
+        String query = ObjectFieldsListQuery(objName);
+        // String g=query.substring(500);
+        // System.debug(g);
+        List<SObject> records = Database.query(query);
+        return records;
+    }
+
+    private static String ObjectFieldsListQuery(String objName){
```

Comment:
nazwy metod z małej litery i bardziej mówiące co konkretna funkcja robi, to znaczy coś w stylu "generateQueryForSObject" i tutaj wgl mozesz wrzucic ten object name i zbieranie jego do środka, bo inicjuesz to na początku a później tego nie używasz nigdy tylko wewnatrz tej metody